TicTacToeConsoleApp/Program.cs

```diff
@@ -0,0 +1,174 @@
+﻿namespace TicTacToeConsoleApp
+{
+    public class Program
+    {
+        private static Dictionary<string, string> board;
+        private static string currentPlayer;
+
+        private static void Main(string[] args)
+        {
+            // Инициализация игрового поля и переменных
+            board = new Dictionary<string, string>();
+
+            currentPlayer = "X";
+
+            InitializeBoard();
+
+            // Основной цикл игры
+            bool gameOver = false;
+            while (!gameOver)
+            {
+                // Вывод игрового поля
+                PrintBoard();
+
+                // Получение хода от текущего игрока
+                bool validMove = false;
+
+                string move = "";
+
+                while (!validMove)
+                {
+                    Console.WriteLine("Ход игрока " + currentPlayer);
+                    Console.WriteLine("Введите координаты клетки в формате 'ряд столбец':");
+
+                    move = Console.ReadLine();
+
+                    validMove = IsValidMove(move);
+
+                    if (!validMove)
+                    {
+                        Console.WriteLine("Некорректный ход! Попробуйте еще раз.");
+                    }
+                }
+                // Обновление игрового поля
+                board[move] = currentPlayer;
+
+                // Проверка на победу
+
+                if (IsWinResult()
+                    || IsDrawResult())
+                {
+                    gameOver = true;
+                    continue;
+                }
+
+                // Смена текущего игрока
+                SwitchPlayer();
+            }
+
+            // Вывод окончательного игрового поля
+            PrintBoard();
+
+            if (IsWinResult())
+                PrintWinner();
+
+            if (IsDrawResult())
+                PrintNonWinner();
+
+            Console.WriteLine("Игра окончена!");
+        }
+
+        // Инициализация игрового поля
+        private static void InitializeBoard()
+        {
+            for (int i = 0; i < 3; i++)
+            {
+                for (int j = 0; j < 3; j++)
+                {
+                    board.Add(i + " " + j, " ");
+                }
+            }
+        }
+
+        private static void PrintWinner()
+        {
+            Console.WriteLine("Игрок " + currentPlayer + " победил!");
+        }
+
+        private static void PrintNonWinner()
+        {
+            Console.WriteLine("Игра окончена вничью!");
+        }
+
+        // Вывод игрового поля на экран
+        private static void PrintBoard()
+        {
+            Console.Clear();
+            Console.WriteLine("   0  1  2");
+            for (int i = 0; i < 3; i++)
```

Comment:
It’s a good idea to include the field size numbers in the configuration or constants so that they are in one place, then use the variable in all methods